compiler/scenario-service/client/src/DA/Daml/LF/ScenarioServiceClient/LowLevel.hs

```diff
@@ -173,12 +174,20 @@ withCheckedProcessCleanup' cp f = withRunInIO $ \run -> bracket
             then return res
             else throwIO $ ProcessExitedUnsuccessfully cp ec
 
-handleCrashingScenarioService :: StreamingProcessHandle -> IO a -> IO a
-handleCrashingScenarioService h act = do
-    res <- race (waitForStreamingProcess h) act
-    case res of
-        Left _ -> fail "Scenario service exited unexpectedly"
-        Right r -> pure r
+handleCrashingScenarioService :: IORef Bool -> StreamingProcessHandle -> IO a -> IO a
+handleCrashingScenarioService exitExpected h act =
+    -- `race` doesn’t quite work here since we might end up
+    -- declaring an expected exit at the very end as a failure.
```

Comment:
Would it be good to point at this PR in that comment? The comment alone doesn't quite do it for me, if there was some clarification or improvement that would be nice. I think I understand what it's getting at, but with my current level of understanding, I think I'd struggle to explain it to others.

Reply:
I expanded the comment and added a link to this PR.

Reply:
That's way clearer - thanks